src/util/codec/mysql/json.rs

```diff
@@ -183,6 +256,70 @@ impl Serialize for Json {
     }
 }
 
+impl PartialEq for Json {
+    fn eq(&self, right: &Json) -> bool {
+        self.partial_cmp(right).unwrap() == Ordering::Equal
+    }
+}
+
+impl PartialOrd for Json {
+    fn partial_cmp(&self, right: &Json) -> Option<Ordering> {
+        let precedence_diff = self.get_precedence() - right.get_precedence();
+        if precedence_diff == 0 {
+            return match self.get_type_code() {
+                TYPE_CODE_LITERAL => {
```

Comment:
It could take the shortcut to skip running the binary comparison between `Json::None`s.